tikv/tikv
# Status server: make CN check more fine-grained

Signed-off-by: Xintao <hunterlxt@live.com>



### What problem does this PR solve?

Issue Number: close #7987 

Problem Summary:
Some query related requests do not need to be CN checked.

### What is changed and how it works?

- Make CN check more fine-grained.
- SecurityConfig is some fields related to exact one StatusServer, let's migrate it into `new()` function.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)
1. create 1tikv/1pd with tls
2. Confirm that it can be accessed by curl with a certificate
3. create 1tikv/1pd with tls and set allowed-cn
4. Confirm that it can be accessed by curl with a right certificate
5. Confirm that it showed "role error" by curl with a wrong certificate

Side effects

No

### Release note 

* Skip CN (common name) check when accessing metrics endpoint when TLS is enabled.
